velox/exec/tests/utils/HiveConnectorTestBase.cpp

```diff
@@ -79,19 +87,41 @@ void HiveConnectorTestBase::writeToFile(
     const std::vector<RowVectorPtr>& vectors,
     std::shared_ptr<dwrf::Config> config,
     const TypePtr& schema) {
-  velox::dwrf::WriterOptions options;
-  options.config = config;
-  options.schema = schema;
   auto localWriteFile = std::make_unique<LocalWriteFile>(filePath, true, false);
   auto sink = std::make_unique<dwio::common::WriteFileSink>(
       std::move(localWriteFile), filePath);
   auto childPool = rootPool_->addAggregateChild("HiveConnectorTestBase.Writer");
-  options.memoryPool = childPool.get();
-  facebook::velox::dwrf::Writer writer{std::move(sink), options};
-  for (size_t i = 0; i < vectors.size(); ++i) {
-    writer.write(vectors[i]);
+  switch (fileFormat_) {
```

Comment:
@minhancao Can you please try if the following code works? This is more or less what we want it to be. We may need to do some refactor first , including the WriterOptions unification. You may need some other touch ups to make this work.
```
void HiveConnectorTestBase::writeToFile(
    const std::string& filePath,
    const std::vector<RowVectorPtr>& vectors,
    dwio::common::FileFormat fileFormat,
    const TypePtr& schema,
    const facebook::velox::dwio::common::WriterOptions& writerOptions
    ) {
  auto localWriteFile = std::make_unique<LocalWriteFile>(filePath, true, false);
  auto sink = std::make_unique<dwio::common::WriteFileSink>(
      std::move(localWriteFile), filePath);
  auto childPool = rootPool_->addAggregateChild("HiveConnectorTestBase.Writer");
  
  std::unique_ptr<dwio::common::Writer> writer = dwio::common::getWriterFactory(fileFormat)
                    ->createWriter(std::move(sink), writerOptions);
  for (size_t i = 0; i < vectors.size(); ++i) {
    writer->write(vectors[i]);
  }
  writer->close();
```

Reply:
@yingsu00 I tried the following code and it works. However, when I ran `velox_hive_iceberg_test` this is the error that caused all 4 tests to failed, do you maybe have any insight on this error?
```
E20240614 15:23:31.796509 16685645 Exceptions.h:67] Line: /Users/minhancao/velox_2/velox/velox/dwio/parquet/reader/ParquetReader.cpp:163, Function:loadFileMetaData, Expression: strncmp(copy.data() + readSize - 4, "PAR1", 4) == 0 No magic bytes found at end of the Parquet file, Source: RUNTIME, ErrorCode: INVALID_STATE
I20240614 15:23:31.796622 16685645 Task.cpp:1871] Terminating task test_cursor 8 with state Failed after running for 2 ms.
I20240614 15:23:31.796690 16685645 Task.cpp:1135] All drivers (1) finished for task test_cursor 8 after running for 2 ms.
unknown file: Failure
C++ exception with description "Exception: VeloxRuntimeError
Error Source: RUNTIME
Error Code: INVALID_STATE
Reason: No magic bytes found at end of the Parquet file
Retriable: False
Expression: strncmp(copy.data() + readSize - 4, "PAR1", 4) == 0
Context: Split [Hive: /tmp/velox_test_b1U82g 0 - 197922] Task test_cursor 8
Additional Context: Operator: TableScan[0] 0
Function: loadFileMetaData
File: /Users/minhancao/velox_2/velox/velox/dwio/parquet/reader/ParquetReader.cpp
Line: 163
```

```
[----------] Global test environment tear-down
[==========] 4 tests from 1 test suite ran. (23277 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 4 tests, listed below:
[  FAILED  ] HiveIcebergTest.positionalDeletesSingleBaseFile
[  FAILED  ] HiveIcebergTest.positionalDeletesMultipleBaseFiles
[  FAILED  ] HiveIcebergTest.baseFileMultiplePositionalDeletes
[  FAILED  ] HiveIcebergTest.positionalDeletesMultipleSplits

 4 FAILED TESTS
 ```

Reply:
Fixed velox_hive_iceberg_test error of missing magic bytes at end of Parquet file. Fix was to supply file format to writeToFile() call in IcebergReadTest.cpp that I missed.